index/generator/library/library.go

```diff
@@ -705,6 +709,54 @@ func validateStackInfo(stackInfo schema.Schema, stackfolderDir string) []error {
 	return errors
 }
 
+// SetLastModifiedValue adds the last modified value to a pre-created index
+// The last modified dates are contained in a file named last_modified.json that is apart of the registry dir
+/* #nosec G304 -- lastModFile is produced from filepath.Join which cleans the input path */
+func SetLastModifiedValue(index []schema.Schema, registryDirPath string) ([]schema.Schema, error) {
+	lastModFile := filepath.Join(registryDirPath, "last_modified.json")
+	bytes, err := os.ReadFile(lastModFile)
+	if err != nil {
+		return index, err
+	}
+
+	var lastModifiedEntries schema.LastModifiedInfo
+	err = json.Unmarshal(bytes, &lastModifiedEntries)
+	if err != nil {
+		return index, err
+	}
+
+	lastModifiedEntriesMap := make(map[string]map[string]string)
+
+	for idx := range lastModifiedEntries.Stacks {
+		updateLastModifiedMap(lastModifiedEntriesMap, &lastModifiedEntries.Stacks[idx])
+	}
+
+	for idx := range lastModifiedEntries.Samples {
+		updateLastModifiedMap(lastModifiedEntriesMap, &lastModifiedEntries.Samples[idx])
+	}
+
+	for i := range index {
+		for j := range index[i].Versions {
+			updateSchemaLastModified(&index[i], j, lastModifiedEntriesMap[index[i].Name][index[i].Versions[j].Version])
+		}
+	}
+
+	return index, nil
+}
+
+func updateLastModifiedMap(m map[string]map[string]string, entry *schema.LastModifiedEntry) {
+	_, ok := m[entry.Name]
+	if !ok {
+		m[entry.Name] = make(map[string]string)
+	}
+
+	m[entry.Name][entry.Version] = entry.LastModified.Format("2006-01-02") //2006-01-02 is used to turn the date into YYYY-MM-DD format
```

Comment:
Shouldn't we include the full datetime here? I think having both date and time can help for sorting purposes in the future.

Reply:
I set it to only `YYYY-MM-DD` as per https://github.com/devfile/api/issues/959#issuecomment-1266046556

Reply:
Hmm, not sure about that, I think it makes more sense to keep all the information. IMO the server should be serving the full info and it's up to the viewer apply a different format.

Reply:
I am okay with the server returning the full info in the response, for the parameter passed to the endpoint however I feel it is best to leave ti in `YYYY-MM-DD` and the server can do the necessary jobs to return the appropriate index. I don't know how much value the endpoint serves if it is looking for precision down to the millisecond to filter, wdyt?